pkg/algorithm/caspool/v1alpha1/manual.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2018 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apisv1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	caspool "github.com/openebs/maya/pkg/caspool/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+/*
+Following is a sample manual CSPC YAML:
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-sparse-pool-test
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-569eb31d-3l88
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      disks:
+      - name: sparse-3c1fc7491f9e4cf50053730740647318
+        id: disk1
+  name: cstor-sparse-pool-test
+  type: sparse
+*/
+
+// GetCasPoolForManualProvisioning returns a CasPool object for manual provisioned pool.
+func (op *Operations) getCasPoolForManualProvisioning() (*apisv1alpha1.CasPool, error) {
+	err := op.validateManual()
+	if err != nil {
+		return nil, errors.Wrapf(err, "validation failed")
+	}
+
+	casPool, err := op.buildCasPoolForManualProvisioning()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to build cas pool object for spc %s", op.CspcObject.Object.Name)
+	}
+	return casPool, nil
+}
+
+// buildCasPoolForManualProvisioning builds a CasPool object for pool creation.
+func (op *Operations) buildCasPoolForManualProvisioning() (*apisv1alpha1.CasPool, error) {
+	nodeNames := op.CspcObject.GetNodeNames()
+	usableNode, diskGroups, err := op.getUsableNode(nodeNames)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get usable node for spc %s", op.CspcObject.Object.Name)
```

Comment:
can we start the error message with:
"failed to build cas pool for manual provisioning: could not get usable node for cspc {%s}", op.CSPCObject.Object.Name

check if `free node` is better than `usable node`? further can this be `eligible node`

I am not too sure of the namings here.